packages/eslint-plugin/tests/rules/return-await.test.ts

```diff
@@ -947,76 +1096,118 @@ async function test<T>(): Promise<T> {
           } catch {}
         }
       `,
-      output: `
+      output: null,
+      errors: [
+        {
+          line: 5,
+          messageId: 'requiredPromiseAwait',
+          suggestions: [
+            {
+              messageId: 'requiredPromiseAwaitSuggestion',
+              output: `
         async function bar() {}
         async function foo() {
           try {
             return await (bar() || undefined || bar());
           } catch {}
         }
       `,
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: `
+async function bar() {}
+async function func1() {
+  try {
+    return null ?? bar();
+  } catch {}
+}
+      `,
+      output: null,
       errors: [
         {
           line: 5,
           messageId: 'requiredPromiseAwait',
+          suggestions: [
+            {
+              messageId: 'requiredPromiseAwaitSuggestion',
+              output: `
+async function bar() {}
+async function func1() {
+  try {
+    return await (null ?? bar());
+  } catch {}
+}
+      `,
+            },
+          ],
         },
       ],
     },
     {
       code: `
-        async function bar() {}
```

Comment:
This just tests 3 totally separable errors, so I broke it up into 3 separate tests.